code/registration.py

```diff
@@ -648,6 +784,14 @@ def find_movie_start_end_empty_frames(
         Tuple of the number of frames to cut from the start and end of the
         movie as (n_trim_start, n_trim_end).
     """
+
+    if isinstance(filepath, str):
+        h5py_name = filepath
+    elif isinstance(filepath, list):
+        array = tiff_to_numpy(filepath)
+        array_to_h5(array, "../scratch/temp.h5")
```

Comment:
it's unnecessary to write the array to disk. just do 
```
n_frames = array.shape[0]
midpoint = n_frames // 2
means = array.mean(axis=(1, 2))
```
and jump forward to line 810 if it's a list of tiffs, something like

```
if isinstance(filepath, str):
    h5py_name = filepath
    with h5py.File(h5py_name, "r") as f:
        n_frames = f[h5py_key].shape[0]
        # We discover empty or extrema frames by comparing the mean of each frames
        # to the mean of the full movie.
        if n_jobs == 1 or n_frames < 2000:
            means = f[h5py_key][:].mean(axis=(1, 2))
        else:
            means = np.concatenate(
                Pool(n_jobs).starmap(
                    _mean_of_batch,
                    product(range(0, n_frames, 1000), [h5py_name], [h5py_key]),
                )
            )
elif isinstance(filepath, list):
    array = tiff_to_numpy(filepath)
    n_frames = array.shape[0]
    means = array.mean(axis=(1, 2))
else:
    raise ValueError("File type not supported")
midpoint = n_frames // 2
mean_of_frames = means.mean()
...
```